ingress/controller.go

```diff
@@ -45,7 +45,11 @@ func (c *controller) Run() {
 	}
 
 	// initialize with current state of ingress
-	c.updateLoadBalancer()
+	err = c.updateLoadBalancer()
```

Comment:
Line 44 starts a new go routine to kick off the watching then this updates the loadbalancer.

Do we want concurrent updates of the load balancer configuration?

I can add a lock inside the load balancer otherwise we need to make sure that the load balancer updates are atomic, but even if they atomic we could end up with a race condition where the initial state (this go routine) gets scheduled after the watch go routing that contains an update. WDYT? 

We should lock in the load balancer then I think we should updateLoadBalancer before starting the watch go routine.